packages/flutter/lib/src/material/carousel.dart

```diff
@@ -905,6 +905,9 @@ class _RenderSliverWeightedCarousel extends RenderSliverFixedExtentBoxAdaptor {
   // (with weight 7), we leave some space before item 0 assuming there is another
   // item -1 as the first visible item.
   int get _firstVisibleItemIndex {
+    if (constraints.viewportMainAxisExtent == 0.0) {
```

Comment:
Can we add a code comment to simply explain why we want to add this check here and below? Something like, If `constraints.viewportMainAxisExtent` is 0,  `firstChildExtent` will be 0 and cause division error.